documentation/modules/cruise-control/con-rebalance-goals-proposals.adoc

```diff
@@ -0,0 +1,180 @@
+Following are the supported goals, which you can reorder in priority, enable, or disable:
+
+. Rack-awareness
+. Replica capacity
+. Capacity
+.. Disk capacity
+.. Network inbound capacity
+.. Network outbound capacity
+.. CPU capacity
+. Replica distribution
+. Potential network output
+. Resource distribution
+.. Disk utilization distribution
+.. Network inbound utilization distribution
+.. Network outbound utilization distribution
+.. CPU utilization distribution
+. Leader bytes-in rate distribution
+. Topic replica distribution
+. Leader replica distribution
+. Preferred leader election
+. Intra-broker disk capacity
+. Intra-broker disk usage distribution    
+
+NOTE: "Write your own" goals, Kafka assigner goals, and Anomaly Detector are not supported in {ProductName}.
+
+For detailed information on each rebalance goal, see link:https://github.com/linkedin/cruise-control/wiki/Pluggable-Components#goals[Goals^] in the Cruise Control Wiki. 
+
+[id="soft-hard-goals"]
+[discrete]
+=== Soft goals and hard goals
+
+The following table explains the differences between soft and hard goals in cluster rebalance calculations:
+
+[cols="3*",options="header",stripes="none",separator=¦]
+|===
+
+¦Rebalance goal type
+¦Description
+¦Example
+
+¦Soft goals
+
+¦Also known as "best effort" goals, soft goals do _not_ need to be satisfied in rebalance proposals. 
+However, they are included in rebalance calculations. 
+A rebalance proposal that satisfies _all_ hard goals but violates one or more soft goals is valid.
+
+Rebalance goals are treated as soft goals, unless designated as hard goals
+
+¦Attempt to distribute a topic's replicas evenly across the cluster (topic replica distribution goal). 
+Ignore this goal if doing so will enable all hard goals to be met.
+
+¦Hard goals
+
+¦Hard goals _must_ be satisfied in rebalance proposals. 
+In other words, a rebalance proposal that does _not_ satisfy all the designated hard goals is rejected by Cruise Control and not sent to the user.
+
+Designating hard goals is optional.
+¦All brokers in the cluster must have fewer than 3 replicas (replica capacity goal).
+
+|===
+
+[discrete]
+=== Default and custom rebalance goals
+
+When interacting with Cruise Control to generate rebalance proposals, you can use the _default rebalance goals_, which are defined on deployment, or set ad hoc _custom rebalance goals_. 
+Both types of rebalance goal can be configured to include hard goals as well as soft goals.  
+
+*Default rebalance goals and cached proposal*
+
+The default rebalance goals are set in the `Kafka` custom resource. 
+You specify a list of goals in `spec.cruiseControl.config.default.goals`.
+
+Unless you set custom rebalance goals, the default rebalance goals are used by Cruise Control to return a _cached rebalance proposal_. 
+The cached proposal reflects the current state of the Kafka cluster and is updated every 15 minutes by default. 
```

Comment:
The cached proposal is generated every 15 min by default using the `default.goals` specified in Kafka spec. If no default goals are specified in the Kafka spec then the `default.goals=goals`. 

Are we documenting what the default `default.goals` are?

Reply:
Added a note:

> If no default optimization goals are specified, the cached proposal is generated using the master goals (specified in Kafka.spec.cruiseControl.config.goals).

You've raised a good point here. So far, I haven't even mentioned the `goals` list, which I think of as the "master goals".